src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -1631,7 +1631,11 @@ void ClassFileParser::parse_fields(const ClassFileStream* const cfs,
     if (parsed_annotations.is_multifield_base() && is_java_primitive(cp->basic_type_for_signature_at(signature_index))) {
       field->field_flags_addr()->update_multifield_base(true);
       char* base_name = cp->symbol_at(name_index)->as_C_string();
-      for (int i = 1; i < parsed_annotations.multifield_arg(); i++) {
+      int bundle_size = parsed_annotations.multifield_arg();
+      if (bundle_size < 0) {
+        bundle_size = VectorSupport::get_max_multifield_count(_class_name);
```

Comment:
An alternative is trying to passing the type info to the `MultiField` annotation, so that the `_class_name` is not needed, and so does to the method `VectorSupport::get_max_multifield_count()`. We can use "max_vector_size(bt)" directly.

Reply:
> An alternative is trying to passing the type info to the `MultiField` annotation, so that the `_class_name` is not needed, and so does to the method `VectorSupport::get_max_multifield_count()`. We can use "max_vector_size(bt)" directly.

Thanks @XiaohongGong , you mean pass explicit vector lane type to multifiled annotation like @multifield(-1,  long.class). Please consider following points 
- We will still need explicit VectorPayload class definitions for mask/shuffule max payloads for different vector types.
- Mask/Shuffle backing storage field type is common for all kinds of vector i.e. byte for shuffle and Boolean for mask.
- We can save on complexity of extending the annotation semantics for max species by doing an efficient vmSymbolID lookup. 
- In future if we refactor shuffle lanes to match the lane types of corresponding vector then additional element type in multifield will be redundant as the backing field type will itself be different for max shuffle payloads.

Reply:
I understand. It's not a problem for mask/shuffle, since we can add the type info for them as well. And I think that's the main reason that we need the type info. Because we can get the type info during class parser for each field of the normal vector classes. Anyway, looking up symbol ID can also work well.